sonar-ballerina-plugin/src/test/java/io/ballerina/sonar/BallerinaSensorTest.java

```diff
@@ -0,0 +1,168 @@
+                StandardCharsets.UTF_8);
+
+        // Append mock settings file
+        Settings settings = context.settings().appendProperty("analyzedResultsPath", filePath.toString());
+        context.setSettings((MapSettings) settings);
+
+
+        // Add mock input files
+        InputFile ballerinaFile = createInputFileFromPath("main.bal");
+        context.fileSystem().add(ballerinaFile);
+
+        // Trigger analysis
+        BallerinaSensor sensor = sensor();
+        sensor.execute(context);
+        if (Files.exists(filePath)) {
+            Files.delete(filePath);
+        }
+        List<Issue> issues = context.allIssues().stream().toList();
+        Assert.assertEquals(issues.size(), 3);
+        assertIssue(issues.get(0), "ballerina:Ballerina1", "Avoid checkpanic",
+                21, 17, 21, 39);
+        assertIssue(issues.get(1), "ballerina:ballerina/example_module_static_code_analyzer:1", "rule 1",
+                17, 0, 22, 1);
+        assertIssue(issues.get(2), "ballerina:ballerinax/example_module_static_code_analyzer:1", "rule 1",
+                17, 0, 22, 1);
+        List<AdHocRule> adHocRules = context.allAdHocRules().stream().toList();
+        Assert.assertEquals(adHocRules.size(), 1);
+        AdHocRule adHocRule = adHocRules.get(0);
+        Assert.assertEquals(adHocRule.severity(), Severity.MAJOR);
+        Assert.assertEquals(adHocRule.type(), RuleType.CODE_SMELL);
+        Assert.assertEquals(adHocRule.name(), "exampleOrg/example_module_static_code_analyzer:1");
+        Assert.assertEquals(adHocRule.description(), "rule 1");
+        Assert.assertEquals(adHocRule.engineId(), "ballerina_external_analyzer");
+        Assert.assertEquals(adHocRule.ruleId(), "exampleOrg/example_module_static_code_analyzer:1");
+        List<ExternalIssue> externalIssues = context.allExternalIssues().stream().toList();
+        Assert.assertEquals(externalIssues.size(), 1);
+        ExternalIssue externalIssue = externalIssues.get(0);
+        Assert.assertEquals(externalIssue.severity(), Severity.MAJOR);
+        Assert.assertEquals(externalIssue.type(), RuleType.CODE_SMELL);
+        Assert.assertEquals(externalIssue.engineId(), "ballerina_external_analyzer");
+        Assert.assertEquals(externalIssue.ruleId(), "exampleOrg/example_module_static_code_analyzer:1");
+        IssueLocation issueLocation = externalIssue.primaryLocation();
+        Assert.assertEquals(issueLocation.message(), "rule 1");
+        TextRange textRange = issueLocation.textRange();
+        Assert.assertNotNull(textRange);
+        Assert.assertEquals(textRange.start().line(), 17);
+        Assert.assertEquals(textRange.start().lineOffset(), 0);
+        Assert.assertEquals(textRange.end().line(), 22);
+        Assert.assertEquals(textRange.end().lineOffset(), 1);
+    }
+
+    @Test(description = "Test the BallerinaSensor with the scan command")
+    void test_ballerina_sensor_with_scan_command() {
```

Comment:
Fix the test name, camelCase.

Reply:
Fixed with f6af3c568b8e52217f98f0ad07ed9bf1fef63061